### ISSUE TITLE
Documenting generated macros

### ISSUE THREAD

**Issue by rushmorem** [dtolnay/proc-macro-hack#12]
I have a macro called `args` [defined here] that is using this crate. I have tried adding documentation for it as shown in your README. However, the documentation is not showing up in the output generated by `cargo doc`. Is this already supported?

[defined here]: https://github.com/rust-rethinkdb/reql/blob/9bb283328075a4bc67672d71ec38e79c51b0dfc5/src/macros.rs#L61

**Comment by dtolnay** [comment]
You are right, that does not work. I will try to publish a fix tonight.

**Comment by rushmorem** [comment]
Awesome. Thanks!

**Comment by dtolnay** [comment]
I fixed this in proc-macro-hack 0.3.3. The only catch is that you will need to move your doc comment into the macro like this. I know that what you wrote looks nicer but I don't think Rust has a way for proc-macro-hack to see it - sorry!

```rust
proc_macro_expr_decl! {
    /// Splice an array of arguments into another term
    ///
    /// ...
    args! => args_impl
}
```

---

![selection_029](https://cloud.githubusercontent.com/assets/1940490/24387060/cbbacf68-1327-11e7-9def-5c416f43ca0e.png)

**Comment by rushmorem** [comment]
> I fixed this in proc-macro-hack 0.3.3.

Thank you :)

> The only catch is that you will need to move your doc comment into the macro like this.

No problem. This is how I actually expected it to be done and had tried that first. I only had it the other way to appease the compiler since the correct way wasn't compiling before your fix.
